core/src/transport/and_then.rs

```diff
@@ -91,31 +78,31 @@ where
 ///
 /// Applies a function to every stream item.
 #[derive(Debug, Clone)]
-pub struct AndThenStream<TListener, TMap> { stream: TListener, listen_addr: Multiaddr, fun: TMap }
+pub struct AndThenStream<TListener, TMap> {
+    stream: TListener,
+    fun: TMap
+}
 
 impl<TListener, TMap, TTransOut, TMapOut, TListUpgr, TTransErr> Stream for AndThenStream<TListener, TMap>
 where
-    TListener: Stream<Item = (TListUpgr, Multiaddr), Error = TTransErr>,
+    TListener: Stream<Item = ListenerEvent<TListUpgr>, Error = TTransErr>,
     TListUpgr: Future<Item = TTransOut, Error = TTransErr>,
-    TMap: FnOnce(TTransOut, ConnectedPoint) -> TMapOut + Clone,
+    TMap: FnOnce(TTransOut, Endpoint) -> TMapOut + Clone,
```

Comment:
What's the rational for changing `ConnectedPoint` to `Endpoint`?

Reply:
The replacement of `ConnectedPoint` by `Endpoint` was a side-effect of some previous iteration of this PR which did not make the listen address available in upgrades. This got eventually corrected, but I forgot to bring back the `ConnectedPoint`.